### DISCUSSION TITLE
Checkbox list chooses all the fields on create

### DISCUSSION THREAD

**Discussion by matijarezek** [filamentphp/filament#12713] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.75

### How can we help you?

I have a problem with form on my user model.
This is the code:

```
CheckboxList::make('position_id')
                        ->relationship(name: 'positions',titleAttribute: 'name')
                        ->hidden(fn($get): bool => !($get('is_admin')))
```

So on create user, when I select one of the 'Positions' it automatically selects all of them, no matter how many there are.
But on edit user I can normally choose one by one.

Is there something I am missing?

**Comment by leandrocfe** [comment]
What kind of relationship is positions?

**Reply by matijarezek** [reply]
User belongs to many positions. Position has many users.
With a pivot table.

**Reply by matijarezek** [reply, marked as answer]
Managed to solve it. 

Changed `CheckboxList::make('position_id')` to `CheckboxList::make('position')`

And now it works as it should. Weird.
